src/toolkits/style_transfer/style_transfer.cpp

```diff
@@ -347,6 +353,92 @@ void style_transfer::infer_derived_options() {
   add_or_update_state({{"training_iterations", 0}});
 }
 
+gl_sframe style_transfer::get_styles(variant_type style_index) {
+  gl_sframe style_sf = read_state<gl_sframe>("styles");
+  gl_sarray style_filter = convert_style_indicies_to_filter(style_index);
+
+  return style_sf[style_filter];
+}
+
+gl_sframe style_transfer::style_sframe_with_index(gl_sarray styles) {
+  std::vector<flexible_type> index_list;
+  flex_int num_styles = read_state<flex_int>("num_styles");
+  index_list.resize(num_styles);
+  std::iota(index_list.begin(), index_list.end(), 0);
+
+  return gl_sframe({
+      {"index", index_list},
+      {"style", styles},
+  });
+}
+
+/**
+ * convert_style_indicies_to_filter
+ *
+ * This function takes a variant type and converts it into a boolean filter. The
+ * elements at the indicies we want to keep are set to a value of `1`, the
+ * elements we don't want to keep are set to a value of `0`.
+ */
+gl_sarray style_transfer::convert_style_indicies_to_filter(const variant_type& data) {
+  // read the `num_styles`
+  flex_int num_styles = read_state<flex_int>("num_styles");
+  if (variant_is<flex_list>(data) || variant_is<flex_vec>(data)) {
+    // if the type is `flex_list` or `flex_vec` create a vector where every
+    // element is set to zero
+    std::vector<flexible_type> index_filter(num_styles, 0);
+    flex_list index_list = variant_get_value<flex_list>(data);
+    // Assert that the list is not zero-length
+    ASSERT_NE(index_list.size(), 0);
+    
+    // populate the indicies that are selected by the flex_list 
+    std::for_each(index_list.begin(), index_list.end(),
+                  [&index_filter, num_styles](flexible_type& ft) {
+                    // assert if the type is an integer or a float
+                    ASSERT_TRUE(ft.get_type() == flex_type_enum::INTEGER ||
+                                ft.get_type() == flex_type_enum::FLOAT);
+
+                    // parse the float or integer value based on the type and,
+                    // set the value at the indicies to 1 indicating the filter
+                    // to be true.
+                    switch (ft.get_type()) {
+                      case flex_type_enum::INTEGER: {
+                        flex_int idx = ft.get<flex_int>();
+                        check_style_index(idx, num_styles);
+                        index_filter[idx] = 1;
+                      }
+                      case flex_type_enum::FLOAT: {
+                        int idx = static_cast<int>(ft.get<flex_float>());
+                        check_style_index(idx, num_styles);
+                        index_filter[idx] = 1;
+                        break;
+                      }
+                      default:
+                        log_and_throw(
+                            "Invalid data type! List should contain either "
+                            "flex_float or flex_int values!");
```

Comment:
I don't think we want `flex_float` or `flex_int` to appear in the error messages being surfaced to users (when validating their inputs). What error message do we have for the old Python implementation?

Reply:
The error we have is `"the `style` variable cannot be parsed"` which I feel is a horrible message. Haha.

Reply:
Honestly, it’s better than exposing C++ type names to users who have no idea what “flex_float” or “flex_int” even are!

Reply:
True, what do you think about the newer message?